bundle/src/main/java/com/adobe/acs/commons/hc/impl/SMTPMailServiceHealthCheck.java

```diff
@@ -0,0 +1,117 @@
+package com.adobe.acs.commons.hc.impl;
+
+import com.day.cq.commons.mail.MailTemplate;
+import com.day.cq.mailer.MessageGateway;
+import com.day.cq.mailer.MessageGatewayService;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.CharEncoding;
+import org.apache.commons.lang.text.StrLookup;
+import org.apache.commons.mail.SimpleEmail;
+import org.apache.felix.scr.annotations.*;
+import org.apache.felix.scr.annotations.Properties;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.apache.sling.hc.api.HealthCheck;
+import org.apache.sling.hc.api.Result;
+import org.apache.sling.hc.util.FormattingResultLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.internet.InternetAddress;
+import java.util.*;
+
+@Component(metatype = true,
+        label = "ACS AEM Commons - E-Mail Service (SMTP) Health Check",
+        description = "Checks if the AEM E-Mail Service can connect and send mail via the configured SMTP server.")
+@Properties({
+        @Property(
+                name = HealthCheck.NAME,
+                value = "SMTP Mail Service",
+                propertyPrivate = true),
+        @Property(
+                label = "Tags",
+                name = HealthCheck.TAGS,
+                unbounded = PropertyUnbounded.ARRAY,
+                value = {"integrations", "smtp", "email"},
+                description = "Tags for this check to be used by composite health checks."),
+        @Property(
+                name = HealthCheck.MBEAN_NAME,
+                value = "smtpMailService",
+                propertyPrivate = true)})
+@Service
+public class SMTPMailServiceHealthCheck implements HealthCheck {
+    private static final Logger log = LoggerFactory.getLogger(SMTPMailServiceHealthCheck.class);
+
+    private static String MAIL_TEMPLATE =
+            System.getProperty("line.separator") +
+            "Sling Health Check for AEM E-mail Service connectivity";
+
+    private static final String DEFAULT_EMAIL = "healthcheck@example.com";
+    @Property(
+            label = "Test E-mail Address",
+            description = "E-mail address to send test message to.",
+            value = DEFAULT_EMAIL)
+    private static final String PROP_EMAIL = "email";
+    private String email;
+
+    @Reference
+    private MessageGatewayService messageGatewayService;
+
+    @Activate
+    protected void activate(Map<String, Object> properties) {
+        this.email = PropertiesUtil.toString(properties.get(PROP_EMAIL), DEFAULT_EMAIL);
+    }
+
+    @Override
+    public Result execute() {
+        final FormattingResultLog resultLog = new FormattingResultLog();
+
+        if (messageGatewayService == null) {
+            resultLog.warn("MessageGatewayService OSGi service could not be found.");
```

Comment:
Although note that it will never be null since the `@Reference` above isn't optional.